.github/scripts/tryrebase.py

```diff
@@ -36,6 +39,39 @@ def rebase_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False) -> None:
                         "git pull --rebase`)", dry_run=dry_run)
 
 
+def rebase_ghstack_onto(pr: GitHubPR, repo: GitRepo, dry_run: bool = False) -> None:
+    if subprocess.run([sys.executable, "-m", "ghstack", "--help"], capture_output=True).returncode != 0:
+        subprocess.run([sys.executable, "-m", "pip", "install", "ghstack"])
+    orig_ref = f"{re.sub(r'/head$', '/orig', pr.head_ref())}"
+    onto_branch = pr.default_branch()
+
+    repo.fetch(orig_ref, orig_ref)
+    repo._run_git("rebase", onto_branch, orig_ref)
+    if dry_run:
+        print("Don't know how to dry-run ghstack")
+    else:
+        push_result = subprocess.run(["ghstack"], capture_output=True).stdout.decode("utf-8")
```

Comment:
Is there a chance this fails? If so, we should communicate to the user the failure message.